Assets/MixedRealityToolkit.Providers/WindowsMixedReality/BaseWindowsMixedRealitySource.cs

```diff
@@ -0,0 +1,495 @@
+{
+    /// <summary>
+    /// A Windows Mixed Reality Controller Instance.
+    /// </summary>
+    public abstract class BaseWindowsMixedRealitySource : BaseController
+    {
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="trackingState"></param>
+        /// <param name="controllerHandedness"></param>
+        /// <param name="inputSource"></param>
+        /// <param name="interactions"></param>
+        public BaseWindowsMixedRealitySource(TrackingState trackingState, Handedness controllerHandedness, IMixedRealityInputSource inputSource = null, MixedRealityInteractionMapping[] interactions = null)
+                : base(trackingState, controllerHandedness, inputSource, interactions)
+        {
+        }
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultLeftHandedInteractions => DefaultInteractions;
+
+        /// <inheritdoc />
+        public override MixedRealityInteractionMapping[] DefaultRightHandedInteractions => DefaultInteractions;
+
+        /// <inheritdoc />
+        public override void SetupDefaultInteractions(Handedness controllerHandedness)
+        {
+            AssignControllerMappings(DefaultInteractions);
+        }
+
+#if UNITY_WSA
+
+        /// <summary>
+        /// The last updated source state reading for this Windows Mixed Reality Source.
+        /// </summary>
+        public InteractionSourceState LastSourceStateReading { get; protected set; }
+
+        private Vector3 currentControllerPosition = Vector3.zero;
+        private Quaternion currentControllerRotation = Quaternion.identity;
+        private MixedRealityPose lastControllerPose = MixedRealityPose.ZeroIdentity;
+        private MixedRealityPose currentControllerPose = MixedRealityPose.ZeroIdentity;
+
+        private Vector3 currentPointerPosition = Vector3.zero;
+        private Quaternion currentPointerRotation = Quaternion.identity;
+        private MixedRealityPose currentPointerPose = MixedRealityPose.ZeroIdentity;
+
+        private Vector3 currentGripPosition = Vector3.zero;
+        private Quaternion currentGripRotation = Quaternion.identity;
+        private MixedRealityPose currentGripPose = MixedRealityPose.ZeroIdentity;
+
+        #region Update data functions
+
+        /// <summary>
+        /// Update the controller data from the provided platform state
+        /// </summary>
+        /// <param name="interactionSourceState">The InteractionSourceState retrieved from the platform</param>
+        public virtual void UpdateController(InteractionSourceState interactionSourceState)
+        {
+            if (!Enabled) { return; }
+
+            UpdateControllerData(interactionSourceState);
+
+            if (Interactions == null)
+            {
+                Debug.LogError($"No interaction configuration for Windows Mixed Reality {ControllerHandedness} Source");
+                Enabled = false;
```

Comment:
Could be existing behavior, but curious how this works out - it looks like if Enabled was false, we would have exited out early on and never updated the interactionSourceState. However if Interactions is null here, we would set Enabled to false, but still update interactionSourceState at the end. Is the rest of this stuff depending on this or does it not really matter?